### PR TITLE
feat: Added cms-mobile routes

### PR DESCRIPTION
**Pull request by ItsFlash10**

### CODE REVIEW

**Comment by devsargam** [comment]
@ItsFlash10 is it done?

**Comment by devsargam** [comment]
LMK when it's ready to review

**Comment by ItsFlash10** [comment]
> @ItsFlash10 are u still working on the mobile app?

@rishavvajpayee will follow up on any comments in this PR

File: src/app/api/mobile/courses/[courseId]/[collectionId]/[contentId]/route.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import db from '@/db';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function GET(req : NextRequest,{ params }: { params: { contentId: string } }) {
+  try {
+    const { contentId } = params;
+    const contents = await db.content.findUnique({
+      where: {
+        id: parseInt(contentId, 10),
```
Comment (SujithThirumalaisamy) [inline src/app/api/mobile/courses/[courseId]/[collectionId]/[contentId]/route.ts:9]: missed check for if a user has purchased that course or not

**Review by SujithThirumalaisamy — CHANGES_REQUESTED** [review]

File: src/app/api/mobile/courses/[courseId]/[collectionId]/[contentId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+export async function GET(req : NextRequest,{ params }: { params: { contentId: string } }) {
+  try {
+    const { contentId } = params;
+    const user = JSON.parse(req.headers.get('g') || '');
+    const userContentAccess = await checkUserContentAccess(user.id, contentId);
+    if (!userContentAccess) {
+      return NextResponse.json({ message: 'User does not have access to this content' }, { status: 403 });
+    }
+    const contents = await db.content.findUnique({
```
Comment (devsargam) [inline src/app/api/mobile/courses/[courseId]/[collectionId]/[contentId]/route.ts:32]: We are already making a db call for finding if the user has the course access or not. Why don't we reuse the query and get the contents from there itself?

https://github.com/code100x/cms/blob/45668f675f430bb0eb6c8490f6a9d76bd397c997/src/app/api/mobile/courses/%5BcourseId%5D/%5BcollectionId%5D/%5BcontentId%5D/route.ts#L4-L22

File: src/app/api/mobile/courses/[courseId]/[collectionId]/route.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    if (!user) {
+      return NextResponse.json({ message: 'User not found' }, { status: 401 });
+    }
+    
+    const { collectionId } = params;
+    const userHasCollectionAccess = await checkUserCollectionAccess(user.id, collectionId);
+    if (!userHasCollectionAccess) {
+      return NextResponse.json({ message: 'User does not have access to this collection' }, { status: 403 });
+    }
+    const collectionData = await db.content.findMany({
```
Comment (devsargam) [inline src/app/api/mobile/courses/[courseId]/[collectionId]/route.ts:40]: Same as above

File: src/app/api/mobile/courses/[courseId]/route.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const { courseId } = params;
+
+    const userCourseAccess = await checkUserCourseAccess(user.id, courseId);
+    if (!userCourseAccess) {
+      return NextResponse.json(
+        { message: 'User does not have access to this course' },
+        { status: 403 },
+      );
+    }
+    const folderContents = await db.content.findMany({
```
Comment (devsargam) [inline src/app/api/mobile/courses/[courseId]/route.ts:35]: Save as above

**Review by devsargam — CHANGES_REQUESTED** [review]

**Pull request merged**